static/js/cf-frontend.js

```diff
@@ -346,6 +368,21 @@ function selectedCheckerOnChange() {
 
 
 }
+
+function linkGen(){
+  var checker_value = $("#type_system").val();
+  var typeURL = encodeURI(checker_value);
+  typeURL = "?type=" + typeURL;
```

Comment:
Using anchor hash `#` is better than using query `?`, because query `?` generally represent some query parameters that front-end would like to query to the back-end.

Also, I think actually this URL decoding/encoding feature could be done by `bbq`, which is a more elegant way (the way used by `pytutor`). I will detail this idea in the overall review comment.